tests/unit/pubsub_v1/subscriber/test_streaming_pull_manager.py

```diff
@@ -561,6 +574,7 @@ def test_close():
     ) = make_running_manager()
 
     manager.close()
+    await_manager_shutdown(manager, timeout=3)
```

Comment:
Why not make this an argument to close?  `manager.close(wait=3)`

Then you wouldn't need await_manager_shutdown, nor the instance variable.

Reply:
There is another PubSub client called PubSub Lite, and it depends on some of the code from here - we try to be conservative with any interface changes, especially because we've just had a [regression](https://github.com/googleapis/python-pubsub/issues/318) when we added a new optional parameter.

We can discuss at the weekly if this particular change is safe to make.

Reply:
My suggestion seems safe, as the omitted argument would behave as existing, but I'm happy to defer.

Reply:
It indeed seems safe, but then again, we also though that the last time. :laughing:
I'll double check tomorrow, added to the agenda.

Reply:
Oh, I should add - this class is internal, it's not exposed to the end users. Users can only trigger shutdown through the [StreamingPullFuture.cancel()](https://github.com/googleapis/python-pubsub/blob/c707f81fdadd118bd2bc48aebe74d453f35cad0d/google/cloud/pubsub_v1/subscriber/futures.py#L46-L51) method.

Now, we cannot touch that, because if the future sends something to the `manager.close()`, it will break PubSub Lite that uses the same future, but a different manager under the hood (i.e. one that does not support that new argument to `close()`).

Reply:
Wouldn't that also be a problem if the future relied on the new method?

Reply:
You mean if `StreamingPullFuture.cancel()` called something else instead, e.g. `self._manager.new_method()`? That would, too, be a problem, because the PubSub Lite's manager would not have that new method.

I might not understand the last question here? The future itself is not meant to be changed, at least not without coordination with PubSub Lite, meaning that changes to `StreamingPullManager.close()` would only serve the tests. Also, `close()` itself should be non-blocking, so that ` StreamingPullFuture.cancel()` calling it is also non-blocking.

Or is the suggestion to add a new for-test-only method to the manager, e.g. `manager.close_with_wait(wait=3)`?
(IMHO the test helper code should be palced in the tests themselves)

Reply:
As discussed in the weekly - we'll add a "blocking close" method to the manager to be used as a helper in the tests, while `StreamingPullManager.close()` itself will remain intact (it needs to be non-blocking, and we cannot change `StreamingPullFuture.cancel()` that invokes it).

@jimfulton Did I summarize that correctly?

Reply:
I guess. :)  I'm worried that the blocking close will duplicate code.  I'm fine with leaving the PR as it is.